storage/resource_file.go

```diff
@@ -14,27 +12,11 @@ import (
 )
 
 func getContentReader(data *schema.ResourceData) (io.ReadCloser, error) {
-	source := data.Get("source").(string)
-	var reader io.ReadCloser
-	var err error
-	if source != "" {
-		reader, err = os.Open(source)
-		if err != nil {
-			return nil, err
-		}
-	}
-	contentBase64 := data.Get("content_base64").(string)
-	if contentBase64 != "" {
-		decodedString, err := base64.StdEncoding.DecodeString(contentBase64)
-		if err != nil {
-			return nil, err
-		}
-		reader = io.NopCloser(bytes.NewReader(decodedString))
-		if err != nil {
-			return nil, err
-		}
+	content, err := workspace.ReadContent(data)
+	if err != nil {
+		return nil, err
 	}
-	return reader, err
+	return io.NopCloser(bytes.NewReader(content)), nil
```

Comment:
With files up to 5Gb supported, reading them completely into memory may not be a good idea.